clang/include/clang/Tooling/DependencyScanning/DependencyScanningWorker.h

```diff
@@ -124,24 +125,28 @@ class DependencyScanningWorker {
                            llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> FS);
 
   /// Run the dependency scanning tool for a given clang driver command-line,
-  /// and report the discovered dependencies to the provided consumer. If \p
-  /// ModuleName isn't empty, this function reports the dependencies of module
-  /// \p ModuleName.
+  /// and report the discovered dependencies to the provided consumer. If
+  /// TUBuffer is not nullopt, it is used as TU input for the dependency
+  /// scanning. Otherwise, the input should be included as part of the
+  /// command-line. ModuleNames is a list of additional modules to be imported
+  /// in addition to what is referenced in TU.
   ///
   /// \returns false if clang errors occurred (with diagnostics reported to
   /// \c DiagConsumer), true otherwise.
-  bool computeDependencies(StringRef WorkingDirectory,
-                           const std::vector<std::string> &CommandLine,
-                           DependencyConsumer &DepConsumer,
-                           DependencyActionController &Controller,
-                           DiagnosticConsumer &DiagConsumer,
-                           std::optional<StringRef> ModuleName = std::nullopt);
+  bool computeDependencies(
+      StringRef WorkingDirectory, const std::vector<std::string> &CommandLine,
+      DependencyConsumer &DepConsumer, DependencyActionController &Controller,
+      DiagnosticConsumer &DiagConsumer,
+      std::optional<std::variant<StringRef, llvm::MemoryBufferRef>>
+          AdditionalInfo = std::nullopt);
```

Comment:
Using `variant` with general purpose types like `StringRef` and `MemoryBufferRef` is confusing. It seems like an overload with a named parameter would be better.

Reply:
Agree. I mostly concern about not to diverge the implementation underneath. I will try refactoring a bit to minimize duplication for overloaded version.